### PR TITLE
HRT-97 Lookup by ExtID

### PR DESCRIPTION
**Pull request by Dizzzmas**
When we are doing `GET` for a single entity, we are using `ExtID` as the key a lot. 

This change will allow us to write in all of our routes this:
```
    def get(self, pk: str) -> Entity:
        return super().get(pk)
```
instead of this:
 ```
def _lookup(self, pk) -> Entity:
        entity = self.model.query.filter_by(extid=pk).one_or_none()
        if not entity:
            abort(404)
        return entity

    def get(self, pk: str) -> Entity:
        return super().get(pk)
```

Also changed the test db to postgres, since sqlite doesn't support our ExtID

### CODE REVIEW

File: smorest_crud/view.py [hunk 1]
```diff
@@ -164,9 +166,15 @@ class ResourceView(CRUDView):
...
     def _lookup(self, pk):
-        """Get model by primary key."""
-        item = self.model.query.get_or_404(pk)
-        return item
+        """Get model by primary key or extid."""
+        if isinstance(pk, int) or str.isdigit(pk):
+            item = self.model.query.get_or_404(pk)
+            return item
+        else:
+            item = self.model.query.filter_by(extid=pk).one_or_none()
```
Comment (revmischa) [inline smorest_crud/view.py:174]: This code is highly-specific to JetKit. `extid` is not a column any other users will have. This isn't suitable for general usage.
Perhaps we should start making our primary keys be UUIDs, or subclass and override this lookup in JetKit.

File: smorest_crud/view.py [hunk 2]
```diff
@@ -164,9 +166,15 @@ class ResourceView(CRUDView):
     delete_enabled: bool = False
 
     def _lookup(self, pk):
-        """Get model by primary key."""
-        item = self.model.query.get_or_404(pk)
-        return item
+        """Get model by primary key or extid."""
```
Comment (revmischa) [inline smorest_crud/view.py:169]: `extid` is not really in scope in this project

**Review by gnudeb — APPROVED** [review]

**Review by revmischa — CHANGES_REQUESTED** [review]
Thanks for doing this! I think the motivation here is great. 
This project shouldn't assume anything about `extid`. Only JetKit knows about that.

File: smorest_crud/view.py (hunk 1 above)
Comment (Dizzzmas) [reply smorest_crud/view.py:174]: Oh, that's a blunder on my side.

Is creating a reusable decorator for the `extid` lookup in JetKit a good idea?

Comment (revmischa) [reply smorest_crud/view.py:174]: Yeah, not sure it needs to be a decorator. We could subclass collection and resource view in JetKit and override `lookup`

Comment (Dizzzmas) [reply smorest_crud/view.py:174]: Roger that

**Comment by Dizzzmas** [comment]
Closing because of the solution being unsuitable for general use

**Pull request closed without merging**